### ISSUE TITLE
Travis doesn't like my code samples

### ISSUE THREAD

**Issue by Sfshaza** [flutter/flutter#8217]
The layout tutorial includes a number of code samples, located in _includes/_code. (By the way, I need to be able to nest these more deeply, so I'd prefer _includes/_code/_tutorials, or something along those lines.

For reference, here's [PR 391](https://github.com/flutter/website/pull/391).

When I created the PR for this tutorial, it failed the Travis checks. Running tools/travis.sh manually yields this:

```
tool/travis.sh 
Path to dart is:
/Users/shaza/Downloads/dart-sdk/bin/dart
Using Dart version:
Dart VM version: 1.23.0-dev.0.0 (Tue Feb 14 14:19:29 2017) on "macos_x64"
tool/travis.sh: line 16: ../flutter/bin/flutter: No such file or directory
```

I assume this is because Travis's environment has commands available in ../flutter, while I do not.

When I look at the errors in the log file, I see:

```
Path to dart is:
../flutter/bin/cache/dart-sdk/bin/dart
Using Dart version:
Dart VM version: 1.22.0-dev.10.6 (Tue Feb  7 12:04:33 2017) on "linux_x64"
Running 'flutter packages get' in website...             |/-\|/-\|/-\|/-\| 8.3s
Run flutter packages get on _includes/_code/card
Running 'flutter packages get' in card...                /-\| 2.2s
Run flutter analyze on _includes/_code/card
Analyzing _includes/_code/card/main.dart...
No analyzer warnings! (ran in 10.9s)
Run flutter packages get on _includes/_code/column
Running 'flutter packages get' in column...              -\|/- 2.3s
Run flutter analyze on _includes/_code/column
Analyzing _includes/_code/column/main.dart...
No analyzer warnings! (ran in 9.0s)
Run flutter packages get on _includes/_code/container
Running 'flutter packages get' in container...           |/-\ 1.8s
Run flutter analyze on _includes/_code/container
Analyzing _includes/_code/container/main.dart...
No analyzer warnings! (ran in 9.1s)
Run flutter packages get on _includes/_code/grid
Running 'flutter packages get' in grid...                -\|/ 1.8s
Run flutter analyze on _includes/_code/grid
Analyzing _includes/_code/grid/main.dart...
No analyzer warnings! (ran in 11.0s)
Run flutter packages get on _includes/_code/hello-world
Running 'flutter packages get' in hello-world...         /-\ 1.5s
Run flutter analyze on _includes/_code/hello-world
Analyzing _includes/_code/hello-world/main.dart...
No analyzer warnings! (ran in 8.7s)
Run flutter packages get on _includes/_code/lakes
Running 'flutter packages get' in lakes...               |/-\ 1.9s
Run flutter analyze on _includes/_code/lakes
Analyzing _includes/_code/lakes/main.dart...
No analyzer warnings! (ran in 8.5s)
Run flutter packages get on _includes/_code/listview
Running 'flutter packages get' in ....                   - 0.8s
Run flutter analyze on _includes/_code/listview
Analyzing _includes/_code/listview/main.dart...
[error] Couldn't infer type parameter 'E'; 'StatelessWidget' must be of type 'ListItem'. (_includes/_code/listview/main.dart, line 31, col 23)
[warning] The element type 'Divider' can't be assigned to the list type 'ListItem'. (_includes/_code/listview/main.dart, line 86, col 3)
(Ran in 7.5s)
The command "./tool/travis.sh" exited with 1.
Done. Your build exited with 1.
```

But when I run dartanalyzer on the *.dart files in _code, it's fine:

```
_code [layout] $ dartanalyzer */*dart
Analyzing [card/main.dart, column/main.dart, container/main.dart, grid/main.dart, hello-world/main.dart, lakes/main.dart, listview/main.dart, pavlova/main.dart, row-expanded-2/main.dart, row-expanded/main.dart, row/main.dart, stack/main.dart, widgets-only/main.dart]...
No issues found
_code [layout] $ 
```
I manually tried `flutter packages get` followed by `flutter analyze` and all is well:

```
_code [layout] $ flutter packages get
Running 'flutter packages get' in website...          0.5s
_code [layout] $ cd ..
_includes [layout] $ flutter analyze */*dart
Analyzing */*dart...
No analyzer warnings! (ran in 0.2s)
```

So, it  worked fine from the command line when I ran `flutter packages get` manually.

I'm not sure what's up!

Also, the travis command is quite slow. When I upload my tutorial, it takes minutes to run the Travis checks. Does this seem right?

**Comment by sethladd** [comment]
Hm, are you using the same version of Dart and dartanalyzer that Travis is using? It's possible that Travis is using a beta SDK, while you are using a stable SDK?

Thank you for using "flutter analyze" when you're testing. That turns all the knobs to 11 and is very strict.

**Comment by Sfshaza** [comment]
I was using dev 1.22 and now I'm using dev 1.23:

```
~ $ dart --version
Dart VM version: 1.23.0-dev.0.0 (Tue Feb 14 14:19:29 2017) on "macos_x64"
~ $ dartanalyzer --version
dartanalyzer version 1.23.0-dev.0.0~ $
```

Oh, this is funny! The dartanalyzer --version command should include a linebreak!!!

**Comment by sethladd** [comment]
Flutter ships its own version of the Dart SDK. You really shouldn't use the Dart tools that come from a random SDK with Flutter. You should use the Flutterified version. "flutter packages get", "flutter analyze". That will ensure you are using the blessed versions of those tools.

So, please do a "flutter upgrade" to ensure you're getting the same version of the Dart SDK that Travis is using (because it also runs "flutter upgrade") and then run "flutter analyze" on your sample locally.

What happens?

**Comment by Sfshaza** [comment]
This has been very informative! I was using the latest dev of Dart, but I didn't realize that I need to use:

```
flutter packages get
flutter analyze
```

When I ran this manually, I got strong mode error messages from one of my examples that had passed though the latest dartanalyzer! (I was running with the latest version of flutter, as I run `flutter upgrade` at least once a day.)

I also had a broken anchor and a broken link, so I've fixed those and Travis is now a happy camper!

Thanks, @sethladd!

(The Travis checks are SUPER slow, though, which concerns me given how small the Flutter site is. We've been suffering slower and slower builds on dartlang and are looking at eliminating the liquid templates to make things faster...)
